src/internal/connector/onedrive/collections.go

```diff
@@ -687,81 +687,52 @@ func (c *Collections) UpdateCollections(
 				continue
 			}
 
-			oneDrivePath, err := path.ToOneDrivePath(collectionPath)
-			if err != nil {
-				return clues.Wrap(err, "invalid path for backup")
-			}
+			// Get the collection for this item.
+			collectionID := ptr.Val(item.GetParentReference().GetId())
 
-			if len(oneDrivePath.Folders) == 0 {
-				// path for root will never change
-				prevCollectionPath = collectionPath
-			} else {
-				prevCollectionPathStr, ok := oldPaths[collectionID]
-				if ok {
-					prevCollectionPath, err = path.FromDataLayerPath(prevCollectionPathStr, false)
-					if err != nil {
-						return clues.Wrap(err, "invalid previous path").With("path_string", prevCollectionPathStr)
-					}
-				}
-
-			}
-
-			col, found := c.CollectionMap[collectionID]
+			collection, found := c.CollectionMap[collectionID]
 			if !found {
-				// TODO(ashmrtn): We should probably tighten the restrictions on this
-				// and just make it return an error if the collection doesn't already
-				// exist. Graph seems pretty consistent about returning all folders on
-				// the path from the root to the item in question. Removing this will
-				// also ensure we always add an entry to get the folder metadata.
-				col = NewCollection(
-					c.itemClient,
-					collectionPath,
-					prevCollectionPath,
-					driveID,
-					c.service,
-					c.statusUpdater,
-					c.source,
-					c.ctrl,
-					invalidPrevDelta,
-				)
-
-				c.CollectionMap[collectionID] = col
-				c.NumContainers++
+				return clues.New("item seen before parent folder").With("item_id", itemID)
```

Comment:
I think we could add the parent's id as well here.
```suggestion
				return clues.New("item seen before parent folder").With("item_id", itemID, "collection_id", collectionID)
```